packages/contracts-bedrock/src/safe/LivenessModule2.sol

```diff
@@ -0,0 +1,323 @@
+    /// @param safe The Safe address that cancelled the challenge.
+    event ChallengeCancelled(address indexed safe);
+
+    /// @notice Emitted when ownership is transferred to the fallback owner.
+    /// @param safe The Safe address that succeeded the challenge.
+    /// @param fallbackOwner The address that claimed ownership if the Safe is unresponsive.
+    event ChallengeSucceeded(address indexed safe, address fallbackOwner);
+
+    /// @notice Semantic version.
+    /// @custom:semver 2.0.0
+    string public constant version = "2.0.0";
+
+    /// @notice Returns challenge_start_time + liveness_response_period if challenge exists, or
+    ///         0 if not.
+    /// @param _safe The Safe address to query.
+    /// @return The challenge end timestamp, or 0 if no challenge.
+    function getChallengePeriodEnd(address _safe) public view returns (uint256) {
+        uint256 startTime = challengeStartTime[_safe];
+        if (startTime == 0) {
+            return 0;
+        }
+        ModuleConfig storage config = safeConfigs[_safe];
+        return startTime + config.livenessResponsePeriod;
+    }
+
+    /// @notice Configures the module for a Safe that has already enabled it.
+    /// @param _config The configuration parameters for the module containing the response
+    ///                period and fallback owner.
+    function configure(ModuleConfig memory _config) external {
+        // Validate configuration parameters to ensure module can function properly.
+        // livenessResponsePeriod must be > 0 to allow time for Safe owners to respond.
+        if (_config.livenessResponsePeriod == 0) {
+            revert LivenessModule2_InvalidResponsePeriod();
+        }
+        // fallbackOwner must not be zero address to have a valid ownership recipient.
+        if (_config.fallbackOwner == address(0)) {
+            revert LivenessModule2_InvalidFallbackOwner();
+        }
+
+        // Check that this module is enabled on the calling Safe.
+        _assertModuleEnabled(msg.sender);
+
+        // Store the configuration for this safe
+        safeConfigs[msg.sender] = _config;
+
+        // Clear any existing challenge when configuring/re-configuring.
+        // This is necessary because changing the configuration (especially
+        // livenessResponsePeriod)
+        // would invalidate any ongoing challenge timing, creating inconsistent state.
+        // For example, if a challenge was started with a 7-day period and we reconfigure to
+        // 1 day, the challenge timing becomes ambiguous. Canceling ensures clean state.
+        // Additionally, a Safe that is able to successfully trigger the configuration function
+        // is necessarily live, so cancelling the challenge also makes sense from a
+        // theoretical standpoint.
+        _cancelChallenge(msg.sender);
+
+        emit ModuleEnabled(msg.sender, _config.livenessResponsePeriod, _config.fallbackOwner);
```

Comment:
Actually, this should be `ModuleConfigured`

Reply:
Fixed in 5be1c2d767b9f6b9ff66edad9497b285d1502d82